jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -510,10 +512,14 @@ ecma_deref_ecma_string (ecma_string_t *string_p) /**< ecma-string */
     case ECMA_STRING_CONTAINER_MAGIC_STRING_EX:
     {
       /* only the string descriptor itself should be freed */
+      break;
+    }
+    case ECMA_STRING_CONTAINER__COUNT:
```

Comment:
I was thinking about this, and realized what is the wrong with adding a default approach. The "default: unreachable" suggests that "there are other valid enums, but they cannot appear here because of certain reasons". But that is not the case here! I consider this COUNT thing a style issue, and I don't want to reduce the readability of the code because of style. I have an idea though, I will update the code.